gdnative-core/src/nativescript/init/property/hint.rs

```diff
@@ -401,3 +412,63 @@ impl ColorHint {
         }
     }
 }
+
+/// Hints for Array, allowing nesting
+#[derive(Debug)]
+pub struct ArrayHint {
+    element_hint: Option<ExportInfo>,
+}
+
+impl ArrayHint {
+    pub fn new() -> Self {
+        ArrayHint { element_hint: None }
+    }
+
+    pub fn with_element_hint<T>(hint: T) -> Self
+    where
+        T: Hint,
+    {
```

Comment:
I don't think this can be used to export arrays of resources, because `Ref<T>` has the hint type of `()`. The constructor would just panic with the current implementation, and would not be able to know which type it's supposed to export even if that is changed.